server/sockets/cohort.js

```diff
@@ -66,12 +70,16 @@ const removeOwnerRoleInCohort = (socket, clients) => {
       });
 
     if (clients.has(userId)) {
-      socket.broadcast
-        .to(clients.get(userId))
-        .emit(CohortActionTypes.REMOVE_OWNER_ROLE_SUCCESS, {
-          ...data,
-          isCurrentUserRoleChanging: true
-        });
+      const { viewId, socketId } = clients.get(userId);
+
+      if (viewId === cohortId) {
+        socket.broadcast
+          .to(socketId)
+          .emit(CohortActionTypes.REMOVE_OWNER_ROLE_SUCCESS, {
+            ...data,
+            isCurrentUserRoleChanging: true
+          });
+      }
```

Comment:
you can do also 
```
const {viewId, socketId} = clients.get(userId);

if(clients.has(userId) && viewId === cohortId ) {
    socket.broadcast()....
.....
```

you'll have one if instead of two nested

Reply:
we have to first check if exist then destructure to certain props as we discussed